src/cargo/util/toml/mod.rs

```diff
@@ -1669,12 +2128,12 @@ impl TomlManifest {
 /// Returns the name of the README file for a `TomlProject`.
 fn readme_for_project(package_root: &Path, project: &TomlProject) -> Option<String> {
     match &project.readme {
-        None => default_readme_from_package_root(package_root),
-        Some(value) => match value {
+        Some(MaybeWorkspace::Defined(value)) => match value {
             StringOrBool::Bool(false) => None,
             StringOrBool::Bool(true) => Some("README.md".to_string()),
             StringOrBool::String(v) => Some(v.clone()),
         },
+        _ => default_readme_from_package_root(package_root),
     }
 }
```

Comment:
It looks like `readme.workspace = true` is causing the default readme to be loaded rather than the workspaces

Reply:
You are correct. I will make this work correctly